@here/harp-mapview/lib/geometry/DisplacedMesh.ts

```diff
@@ -66,42 +58,69 @@ export class DisplacedMesh extends THREE.Mesh {
-     * @param [geometry] Original geometry to displace.
-     * @param [material] Material(s) to be used by the mesh. All must have the same displacement
-     * map.
+     * @param [m_raycastStrategy] Function that will be used to find ray intersections. If not
+     * provided, THREE.Mesh's raycast will be used.
      */
     constructor(
+        geometry: THREE.BufferGeometry,
+        material: THREE.Material | THREE.Material[],
         private m_getDisplacementRange: () => DisplacementRange,
-        geometry?: THREE.Geometry | THREE.BufferGeometry,
-        material?: THREE.Material | THREE.Material[]
+        private m_raycastStrategy?: (
+            mesh: THREE.Mesh,
+            raycaster: THREE.Raycaster,
+            intersects: THREE.Intersection[]
+        ) => void
     ) {
         super(geometry, material);
-        assert(!this.geometry || isBufferGeometry(this.geometry));
-        assert(!this.material || isDisplacementMaterial(this.firstMaterial));
     }
 
-    raycast(raycaster: THREE.Raycaster, intersects: THREE.Intersection[]): void {
-        if (!this.geometry || !this.material) {
-            return;
+    // HARP-9585: Override of base class method, however tslint doesn't recognize it as such.
+    // tslint:disable-next-line: explicit-override
+    get geometry(): THREE.BufferGeometry {
+        return super.geometry as THREE.BufferGeometry;
+    }
+
+    // HARP-9585: Override of base class method, however tslint doesn't recognize it as such.
+    // tslint:disable-next-line: explicit-override
+    set geometry(geometry: THREE.BufferGeometry) {
+        super.geometry = geometry;
+    }
+
+    // HARP-9585: Override of base class method, however tslint doesn't recognize it as such.
+    // tslint:disable-next-line: explicit-override
+    get material(): THREE.Material | THREE.Material[] {
+        return super.material;
+    }
+
+    // HARP-9585: Override of base class method, however tslint doesn't recognize it as such.
+    // tslint:disable-next-line: explicit-override
+    set material(material: THREE.Material | THREE.Material[]) {
+        super.material = material;
+        const firstMaterial = this.firstMaterial;
+        if (isDisplacementMaterial(firstMaterial)) {
+            this.m_displacementMaterial = firstMaterial;
         }
+    }
 
+    // HARP-9585: Override of base class method, however tslint doesn't recognize it as such.
+    // tslint:disable-next-line: explicit-override
+    raycast(raycaster: THREE.Raycaster, intersects: THREE.Intersection[]): void {
         // All materials in the object are expected to have the same displacement map.
-        const material = this.firstMaterial;
-
-        // Use default raycasting implementation if some type is unexpected.
-        if (
-            !isBufferGeometry(this.geometry) ||
-            !isDisplacementMaterial(material) ||
-            !isDataTextureMap(material.displacementMap)
```

Comment:
I've done changes that make checking those contraints here unnecessary:

- isBufferGeometry() is not needed anymore since now we only take THREE.BufferGeometry as parameter in the constructor and setter.
- isDisplacementMaterial() is checked in the constructor and setter, so no need to check it every time raycast is called.